registry/storage/driver/gcs/gcs_test.go

```diff
@@ -69,14 +71,21 @@ func init() {
 		ts = jwtConfig.TokenSource(dcontext.Background())
 	}
 
+	gcs, err := storage.NewClient(context.Background(), option.WithCredentialsJSON([]byte(credentials)))
```

Comment:
This wants the file contents, not the file name. There is another call that should also be using the contents higher up (line 59) so we should probably just read the file and pass it to both. Something like this just after the skip:

```
    jsonKey, err := ioutil.ReadFile(credentials)
    if err != nil {
        panic(err)
    }
```

Reply:
Yes, this won't work, but now that we are at this, we should probably change the env var `GOOGLE_APPLICATION_CREDENTIALS` to something that fits in better with the registry [storage driver config "schema"](https://github.com/distribution/distribution/blob/main/docs/configuration.md#list-of-configuration-options) e.g. `REGISTRY_STORAGE_GCS_CREDENTIALS_PATH` or `REGISTRY_STORAGE_GCS_CREDENTIALS_APPLICATION` and update the config README so it's clear this env var contains the path to creds.

Reply:
But `GOOGLE_APPLICATION_CREDENTIALS` is the exact variable used for discovery by the SDK

Reply:
Oof. Right. Clearly I'm not a GCP user 😬. Let me disappear from this thread.

Reply:
Nooo, then who's going to approve anything? ;)